### DISCUSSION TITLE
Implementing a const table of values?

### DISCUSSION THREAD

**Discussion by ghost** [micropython/micropython#14189] [Other]
- row_offset: tuple = const((0x00, 0x40, 0x14, 0x54)) is that valid?
-  mutable object inside a const(variable) or const(constant) which one is valid
- 2D list or tuple inside a const() function is that valid?
- example: 
```python
# Channel step registers lookup table
__CH_STEPS: tuple[tuple[int, int], ...] = const(
    (
        (0x80, 0x90),  # Channel No. 1: -10dB/step, -1dB/step
        (0x40, 0x50),  # Channel No. 2: -10dB/step, -1dB/step
        (0x00, 0x10),  # Channel No. 3: -10dB/step, -1dB/step
        (0x20, 0x30),  # Channel No. 4: -10dB/step, -1dB/step
        (0x60, 0x70),  # Channel No. 5: -10dB/step, -1dB/step
        (0xA0, 0xB0),  # Channel No. 6: -10dB/step, -1dB/step
    )
)
```
is that valid

**Comment by ned-pcs** [comment]
Did you test it? Your example works fine. But you can't have mutable objects in a `const`; they have to be available to the compiler as a constant expression.
